src/code_80005FD0.c

```diff
@@ -4288,53 +4288,52 @@ void func_80011EC0(s32 arg0, Player *player, s32 arg2, UNUSED u16 arg3) {
     }
 }
 
-void func_800120C8(void) {
+#define GET_PATH_LENGTH(waypoint) for (i = 0; ; i++) { \
+        if ((u16)waypoint[i].posX == 0x8000) { \
+            break; \
+        } \
+    }
+
+void load_vehicles_waypoint_train(void) {
     s32 i;
-    PathNoY *temp;
+    Path2D *temp;
     TrackWaypoint *waypoint = (TrackWaypoint *) VIRTUAL_TO_PHYSICAL2(
-        gSegmentTable[SEGMENT_NUMBER2(d_course_kalimari_desert_track_unknown_waypoints)]
-                    + SEGMENT_OFFSET(d_course_kalimari_desert_track_unknown_waypoints));
+        gSegmentTable[SEGMENT_NUMBER2(d_course_kalimari_desert_waypoint_vehicles_train)]
+                    + SEGMENT_OFFSET(d_course_kalimari_desert_waypoint_vehicles_train));
 
-    for (i = 0; ; i++) {
-        if ((u16)waypoint[i].posX == 0x8000) {
-            break;
-        }
-    }
+    GET_PATH_LENGTH(waypoint)
 
-    temp = D_80163598;
-    D_8016359C = func_8001168C(temp, waypoint, i - 1);
+    temp = gVehicle2DWaypoint;
+    gVehicle2DWaypointLength = generate_2d_path(temp, waypoint, i - 1);
     D_80162EB0 = func_802AE1C0(temp[0].x, 2000.0f, temp[0].z);
 }
 
-void func_80012190(void) {
-    TrackWaypoint *tree;
+void load_vehicles_waypoint_ferry(void) {
```

Comment:
generate_ferry_waypoints.

There's no loading going on here. All that was dma'd somewhere else.